### ISSUE TITLE
no-hooks: provide option to choose which hooks to allow

### ISSUE THREAD

**Issue by jmerrifield** [lo1tuma/eslint-plugin-mocha#110]
It would be great if the `no-hooks` rule accepted options specifying which hooks to allow. Specifically I'd like to be able to allow all hooks except `before`.

Would you accept a PR for this functionality?

**Comment by jfmengels** [comment]
This sounds good to me, as this was pretty much requested in https://github.com/lo1tuma/eslint-plugin-mocha/pull/101 too.

Would a `allow` be fine with you @jmerrifield?

**Comment by jmerrifield** [comment]
That sounds great, so in my case I would configure it like:

```js
rules: {
  'mocha/no-hooks': ['error, { allow: ['beforeEach', 'after', 'afterEach'] }]
}
```

If so, that would be ideal 👍
